main.go

```diff
@@ -214,34 +210,46 @@ func setupControllers(ctx context.Context, mgr ctrl.Manager, cCache *cache.Cache
 		os.Exit(1)
 	}
 
-	crds := findCustomResources(ctx, mgr)
-
 	opts := []jobframework.Option{
 		jobframework.WithManageJobsWithoutQueueName(manageJobsWithoutQueueName),
 		jobframework.WithWaitForPodsReady(waitForPodsReady(cfg)),
 		jobframework.WithKubeServerVersion(serverVersionFetcher),
 	}
 	err := jobframework.ForEachIntegration(func(name string, cb jobframework.IntegrationCallbacks) error {
 		log := setupLog.WithValues("jobFrameworkName", name)
-		if isFrameworkEnabled(cfg, name) && crds.Has(name) {
-			if err := cb.NewReconciler(
-				mgr.GetClient(),
-				mgr.GetEventRecorderFor(fmt.Sprintf("%s-%s-controller", name, constants.KueueName)),
-				opts...,
-			).SetupWithManager(mgr); err != nil {
-				log.Error(err, "unable to create controller")
-				return err
-			}
-			if err := cb.SetupWebhook(mgr, opts...); err != nil {
-				log.Error(err, "Unable to create webhook")
-				return err
-			}
-		} else {
-			if err := noop.SetupWebhook(mgr, cb.JobType); err != nil {
-				log.Error(err, "Unable to create noop webhook")
-				return err
+
+		if isFrameworkEnabled(cfg, name) {
+			if _, err := mgr.GetRESTMapper().RESTMapping(cb.GVK.GroupKind(), cb.GVK.Version); err != nil {
+				// TODO: If the below PR is released, we need to change a way to check if the GVK is registered.
```

Comment:
Actually, one more thought. There is a risk that once the PR is released, say in controller-runtime 0.16, we may not remember to check this when bumping the version and break this. I think it is unlikely a person doing a bump of controller-runtime would look at this comment without prior knowledge. 
Some ideas to mitigate this risk:
1. prepare the check in advance, `if !meta.IsNoMatchError(err) && !errors.As(err, &NoMatchingErr) { return err }`. Then leave the TODO comment to cleanup up once released
2. coordinate with the release of the PR and bump controller-runtime (but might be not necessarily involving)
3. create an issue in advance in kueue to increase visibility, giving a title like "Bump controller-runtime and adjust RESTMapper usage"

I'm leaning towards (1.) or (2.), but (2.) only if it is to happen with Kueue 0.5 release, because I don't like keeping PRs in a freezer for long :). (1.) seems safe to do. 

WDYT?

Reply:
1 sounds good, but also leave a TODO and issue.

Reply:
@mimowo It's a great suggestion. I think `1` would be nice.

> 1 sounds good, but also leave a TODO and issue.

I agree. We should create an issue.

Reply:
Changed logic and created an issue: https://github.com/kubernetes-sigs/kueue/issues/1054